src/main/java/de/tum/in/www1/artemis/repository/specs/UserSpecs.java

```diff
@@ -0,0 +1,225 @@
+     *
+     * @param authorities provided authorities
+     * @param courseIds a set of courseIds which the users need to match
+     * @return specification used to chain database operations
+     */
+    public static Specification<User> getAuthoritySpecification(Set<String> authorities, Set<Long> courseIds) {
+        if (authorities.contains(FILTER_NO_AUTHORITY)) {
+            // Empty authorities
+            return getAllUsersMatchingEmptyAuthorities();
+        }
+        else if (!authorities.isEmpty() && (courseIds.isEmpty() || courseIds.contains(FILTER_EMPTY_COURSES))) {
+            // Match all authorities
+            return getAllUsersMatchingAuthorities(authorities);
+        }
+        return null;
+    }
+
+    /**
+     * Creates the specification to match the state of the user (internal or external).
+     *
+     * @param internal true if the account should be internal
+     * @param external true if the account should be external
+     * @return specification used to chain database operations
+     */
+    public static Specification<User> getInternalOrExternalSpecification(boolean internal, boolean external) {
+        if (!internal && !external) {
+            return null;
+        }
+        else {
+            return (root, query, criteriaBuilder) -> {
+                Predicate internalPredicate = criteriaBuilder.equal(root.get(User_.IS_INTERNAL), internal);
+                Predicate externalPredicate = criteriaBuilder.notEqual(root.get(User_.IS_INTERNAL), external);
+
+                return criteriaBuilder.and(internalPredicate, externalPredicate);
+            };
+        }
+    }
+
+    /**
+     * Creates the specification to match the state of the user (activated or deactivated).
+     *
+     * @param activated   true if the account should be activated
+     * @param deactivated true if the account should be deactivated
+     * @return specification used to chain database operations
+     */
+    public static Specification<User> getActivatedOrDeactivatedSpecification(boolean activated, boolean deactivated) {
+        if (!activated && !deactivated) {
+            return null;
+        }
+        else {
+            return (root, query, criteriaBuilder) -> {
+                Predicate activatedPredicate = criteriaBuilder.equal(root.get(User_.ACTIVATED), activated);
+                Predicate deactivatedPredicate = criteriaBuilder.notEqual(root.get(User_.ACTIVATED), deactivated);
+
+                return criteriaBuilder.and(activatedPredicate, deactivatedPredicate);
+            };
+        }
+    }
+
+    /**
+     * Creates the specification to match the users course.
+     *
+     * @return specification used to chain database operations
+     */
+    public static Specification<User> getAllUsersMatchingEmptyCourses() {
+        return (root, query, criteriaBuilder) -> criteriaBuilder.isEmpty(root.get(User_.GROUPS));
+    }
+
+    /**
+     * Creates the specification to match the users course.
```

Comment:
```suggestion
     * Creates the specification to find all users that are part of any of the given courses.
```
Was copy-pasted from the other method.